### ISSUE TITLE
"No such file or directory" and "Failed to upload R-Code"

### ISSUE THREAD

**Issue by pocketstoosumo** [tsaibermelon/gg-struggle#15]
Whenever I open gg-struggle this pops up: 

`[2021-07-19T11:57:44.880] [WARN] default - [DB] Error reading request 9d9a45e5a3c60c7197fa3922b939cb6a5bdd47d5: Error: ENOENT: no such file or directory, open 'C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/9d9a45e5a3c60c7197fa3922b939cb6a5bdd47d5.ggResp.dump'`

`[2021-07-19T11:57:44.881] [WARN] default - [DB] Error reading request 425a3c24e7030ec0669324a6a1a37c69eed8592e: Error: ENOENT: no such file or directory, open 'C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/425a3c24e7030ec0669324a6a1a37c69eed8592e.ggResp.dump'`

`[2021-07-19T11:57:44.882] [WARN] default - [DB] Error reading request 5e10b3036695e88f4f41e5b191c7c2a3e284ff71: Error: ENOENT: no such file or directory, open 'C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/5e10b3036695e88f4f41e5b191c7c2a3e284ff71.ggResp.dump'`

`[2021-07-19T11:57:44.883] [WARN] default - [DB] Error reading request b529c6abdcbf1cfb914b5e765bc59542c78d2de9: Error: ENOENT: no such file or directory, open 'C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/b529c6abdcbf1cfb914b5e765bc59542c78d2de9.ggResp.dump'`

`[2021-07-19T11:57:44.884] [WARN] default - [DB] Error reading request d5f3af1d575dc11af8db1239a680f35da99d4c70: Error: ENOENT: no such file or directory, open 'C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/d5f3af1d575dc11af8db1239a680f35da99d4c70.ggResp.dump'`

`[2021-07-19T11:57:44.885] [WARN] default - [DB] Error reading request 0741fd8583402ef09f4c615f77bcf2e4dc55140b: Error: ENOENT: no such file or directory, open 'C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/0741fd8583402ef09f4c615f77bcf2e4dc55140b.ggResp.dump'`

`[2021-07-19T11:57:44.886] [WARN] default - [DB] Error reading request 28ef016d2d1639ef1d864cdd126590fe6322438c: Error: ENOENT: no such file or directory, open 'C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/28ef016d2d1639ef1d864cdd126590fe6322438c.ggResp.dump'`

and when I run gg-struggle with Strive it says "Failed to upload R-Code" and closes gg-struggle

Video for reference: https://streamable.com/q7i51p

**Comment by cybermelons** [comment]
try deleting your `%temp%/gg-struggle/gg-struggle.db`. These files are safe to delete.

As far as those errors go, they're not really errors but warnings so you can safely ignore those

**Comment by pocketstoosumo** [comment]
deleting that file removed the warnings but now whenever I load in it doesn't load at all and jumps to "The network features are not available due to not being connected to the network" and closes gg-struggle

**Comment by cybermelons** [comment]
can you run gg-struggle.exe in powershell? that way it’ll stay open after it crashes and we can see the error

Win+R > powershell.exe > `cd $env.programfiles\gg-struggle` > `gg-struggle.exe`

**Comment by pocketstoosumo** [comment]
Your exact instructions didn't work so I did: ``cd "C:\Program Files (x86)\gg-struggle"`` > ``.\gg-struggle.exe``

Here's what it gave me

```
[PROXY] No config specified. Using default ./local.json
[2021-07-19T16:01:42.212] [INFO] default - [PROXY] Logging to C:\Users\nartd\AppData\Local\Temp/gg-struggle/all.log
[2021-07-19T16:01:42.216] [INFO] default - [PROXY] Using pfx file ./gg-struggle-cert.pfx
[2021-07-19T16:01:42.219] [INFO] default - [DB] Loading entries C:\Users\nartd\AppData\Local\Temp/gg-struggle/gg-struggle.db
[2021-07-19T16:01:42.246] [INFO] default - [PROXY] Listening on 443
[2021-07-19T16:01:42.252] [INFO] default -
[2021-07-19T16:01:53.538] [INFO] default - [PROXY] /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
[2021-07-19T16:01:53.539] [INFO] default - [DB] Dumping request to C:\Users\nartd\AppData\Local\Temp/gg-struggle/dumps/b627588615559e35366585b10f1b5e2e203a4a7f.gameReq.dump
[2021-07-19T16:01:53.542] [INFO] default - [CACHE] Hit: /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
_http_server.js:259
    throw new ERR_HTTP_INVALID_STATUS_CODE(originalStatusCode);
    ^

RangeError [ERR_HTTP_INVALID_STATUS_CODE]: Invalid status code: null
    at ServerResponse.writeHead (_http_server.js:259:11)
    at C:\snapshot\server\gg-struggle.js:487:18
    at CacheLayer.get (C:\snapshot\server\gg-struggle.js:68:7)
    at IncomingMessage.<anonymous> (C:\snapshot\server\gg-struggle.js:486:22)
    at IncomingMessage.emit (events.js:388:22)
    at endReadableNT (internal/streams/readable.js:1336:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  code: 'ERR_HTTP_INVALID_STATUS_CODE'
```

**Comment by cybermelons** [comment]
ah, that’s perfect. looks like proxy isn’t properly handling the response from the real gg server.

I’m unsure what the exact error is, but we can at least catch the error and report it wihtout crashing.

**Comment by pocketstoosumo** [comment]
do you know how to fix it?

**Comment by cybermelons** [comment]
yeah, but unfortunately I have to code the fix in. no hotfixes on the user-end. sorry :(

**Comment by pocketstoosumo** [comment]
damn, thanks for helping out

**Comment by cybermelons** [comment]
Yup. Uninstall if it ain’t working for now

**Comment by Pushigoh** [comment]
Changing the IP in the host file to 35.72.213.199 fixed this problem for me!

**Comment by pocketstoosumo** [comment]
Doing that "fixed" the problem. I loaded in but nothing happened on the gg-struggle window and it didn't make the load times faster

**Comment by Jensegaense** [comment]
> Changing the IP in the host file to 35.72.213.199 fixed this problem for me!

Oh my God thank you SO much this fixed it perfectly for me. I'm down from 16 minutes of "Connecting To Network" down to 15 SECONDS now.

**Comment by cybermelons** [comment]
> Doing that "fixed" the problem. I loaded in but nothing happened on the gg-struggle window and it didn't make the load times faster

First load is usually gonna be slow, cause it needs to preload initial requests.

Closing cause the IP issues is fixed in #20 . Please reopen a new issue if this sitll persists
